cdap-standalone/src/main/java/co/cask/cdap/TrackerAppCreationService.java

```diff
@@ -0,0 +1,138 @@
+import co.cask.cdap.client.ProgramClient;
+import co.cask.cdap.client.config.ClientConfig;
+import co.cask.cdap.client.config.ConnectionConfig;
+import co.cask.cdap.common.ArtifactNotFoundException;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.conf.KafkaConstants;
+import co.cask.cdap.common.utils.Tasks;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.proto.artifact.ArtifactSummary;
+import com.google.common.util.concurrent.AbstractExecutionThreadService;
+import com.google.inject.Inject;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Create Tracker App in Default Namespace and start its Programs during Standalone startup.
+ */
+public class TrackerAppCreationService extends AbstractExecutionThreadService {
+
+  private final CConfiguration cConf;
+  private final ClientConfig clientConfig;
+  private ArtifactSummary artifactSummary;
+
+  @Inject
+  public TrackerAppCreationService(CConfiguration cConf) {
+    this.cConf = cConf;
+    String address = cConf.get(Constants.Router.ADDRESS);
+    int routerPort = cConf.getInt(Constants.Router.ROUTER_PORT);
+    this.clientConfig = ClientConfig.builder().setConnectionConfig(
+      ConnectionConfig.builder().setHostname(address).setPort(routerPort).build()).build();
+  }
+
+  @Override
+  protected void run() throws Exception {
+    // Wait until tracker artifact is available
+    final ArtifactClient client = new ArtifactClient(clientConfig);
+    Tasks.waitFor(true, new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        List<ArtifactSummary> artifacts = null;
+        try {
+          artifacts = client.listVersions(Id.Namespace.DEFAULT, "tracker", ArtifactScope.SYSTEM);
+        } catch (ArtifactNotFoundException ex) {
+          // expected, so suppress it
+        }
+        return artifacts != null && artifacts.size() > 0;
+      }
+    }, 5, TimeUnit.MINUTES, 5, TimeUnit.SECONDS, "Waiting for Tracker Artifact to be listed.");
+    // Find the latest tracker artifact based on the version
+    List<ArtifactSummary> artifacts = client.listVersions(Id.Namespace.DEFAULT, "tracker", ArtifactScope.SYSTEM);
+    artifactSummary = artifacts.remove(0);
+    for (ArtifactSummary artifact : artifacts) {
+      ArtifactVersion old = new ArtifactVersion(artifactSummary.getVersion());
+      ArtifactVersion current = new ArtifactVersion(artifact.getVersion());
+      if (current.compareTo(old) > 0) {
+        artifactSummary = artifact;
+      }
+    }
+    createAndStartTracker();
+  }
+
+  private void createAndStartTracker() throws Exception {
+    ApplicationClient appClient = new ApplicationClient(clientConfig);
+    ProgramClient programClient = new ProgramClient(clientConfig);
```

Comment:
Same here. Don't use any client. Should be using the `ApplicationLifecycleService` and `ProgramLifecycleService`.